.pre-commit-hooks.yaml

```diff
@@ -0,0 +1,8 @@
+- id: licensing
+  name: Add license headers to source files
+  description: This script adds license headers to source files in the repository
+  entry: license-files
+  language: python
+  pass_filenames: false
+  require_serial: true
```

Comment:
Why do we need to require_serial?

Reply:
Because this hook potentially modifies files. If it were to do that while `flake8` or `mypy` was also running it would cause weird pre-commit failures.

Reply:
Is that the design we want the script to have? Shouldn't we accept filenames so we can run parallel? If we think that's worthwhile, maybe its an item for the backlog or something to fix now if it wouldn't take too long.

Reply:
I think it's something we should put on the backlog. It's not a quick fix for this script.